ginga/misc/plugins/LineProfile.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+
+class LineProfile(GingaPlugin.LocalPlugin):
+
+    def __init__(self, fv, fitsimage):
+        super(LineProfile, self).__init__(fv, fitsimage)
+
+        self.layertag = 'lineprofile-canvas'
+
+        self.gui_up = False
+
+    def build_gui(self, container):
+        top = Widgets.VBox()
+        top.set_border_width(4)
+
+        vbox, sw, orientation = Widgets.get_oriented_box(container)
+        vbox.set_margins(4, 4, 4, 4)
+        vbox.set_spacing(2)
+
+        self.msgFont = self.fv.getFont("sansFont", 12)
+        tw = Widgets.TextArea(wrap=True, editable=False)
+        tw.set_font(self.msgFont)
+        self.tw = tw
+
+        fr = Widgets.Frame("Instructions")
+        vbox2 = Widgets.VBox()
+        vbox2.add_widget(tw)
+        vbox2.add_widget(Widgets.Label(''), stretch=1)
+        fr.set_widget(vbox2)
+        vbox.add_widget(fr, stretch=0)
+
+        self.plot = Plot.Plot(self.logger, width=2, height=4, dpi=100)
+        ax = self.plot.add_axis()
+        ax.grid(False)
+
+        w = Widgets.wrap(self.plot.get_widget())
+        vbox.add_widget(w, stretch=1)
+
+        # Add a spacer to stretch the rest of the way to the end of the
+        # plugin space
+        spacer = Widgets.Label('')
+        vbox.add_widget(spacer, stretch=1)
+
+        # scroll bars will allow lots of content to be accessed
+        top.add_widget(sw, stretch=1)
+
+        # A button box that is always visible at the bottom
+        btns = Widgets.HBox()
+        btns.set_spacing(3)
+
+        # Add a close button for the convenience of the user
+        btn = Widgets.Button("Close")
+        btn.add_callback('activated', lambda w: self.close())
+        btns.add_widget(btn, stretch=0)
+        btns.add_widget(Widgets.Label(''), stretch=1)
+        top.add_widget(btns, stretch=0)
+
+        # Add our GUI to the container
+        container.add_widget(top, stretch=1)
+        self.gui_up = True
+
+    def instructions(self):
+        self.tw.set_text("""Pick a point""")
+
+    def close(self):
+        chname = self.fv.get_channelName(self.fitsimage)
+        self.fv.stop_local_plugin(chname, str(self))
+        self.gui_up = False
+        return True
+        
+    def start(self):
+        self.instructions()
+
+        self.resume()
+
+    def pause(self):
+        pass
+
+    def resume(self):
+        self.redo()
+        
+    def stop(self):
+        pass
+        
+    def redo(self):
+        image = self.fitsimage.get_image()
+        path = image.get('path', None)
+        fits_f = pyfits.open(path, 'readonly')
+
+        self.lp_plot(fits_f)
+
+    def lp_plot(self, fits_f):
+        wav = self.get_axis(fits_f[7].header)
+        self.plot.plot(wav, fits_f[7].data[156,153,:])
+        title = fits_f[0].header.get('TDESC7') + ' ' + str(fits_f[0].header.get('TWMIN7')) + '-' + str(fits_f[0].header.get('TWMAX7')) + ' Ang'
+        self.plot.set_titles(xtitle='Wavelength', title=title)
+
```

Comment:
Yeah, the line profile position has also got to be selectable.
On 13 Aug 2015 2:06 am, "ejeschke" <notifications@github.com> wrote:

> In ginga/misc/plugins/LineProfile.py
> <https://github.com/ejeschke/ginga/pull/146#discussion_r36943247>:
>
> > +    def stop(self):
> > +        pass
> > +
> > +    def redo(self):
> > +        image = self.fitsimage.get_image()
> > +        path = image.get('path', None)
> > +        fits_f = pyfits.open(path, 'readonly')
> > +
> > +        self.lp_plot(fits_f)
> > +
> > +    def lp_plot(self, fits_f):
> > +        wav = self.get_axis(fits_f[7].header)
> > +        self.plot.plot(wav, fits_f[7].data[156,153,:])
> > +        title = fits_f[0].header.get('TDESC7') + ' ' + str(fits_f[0].header.get('TWMIN7')) + '-' + str(fits_f[0].header.get('TWMAX7')) + ' Ang'
> > +        self.plot.set_titles(xtitle='Wavelength', title=title)
> > +
>
> @rupak0577 <https://github.com/rupak0577>, This plugin looks very tightly
> tied to a particular kind of FITS file. Nothing wrong with that, but I
> think in the current form it may be better as a custom plugin. I think this
> may be some additional impetus to solve issue #92
> <https://github.com/ejeschke/ginga/issues/92>, so that such plugins can
> be distributed separately and installed.
>
> Maybe what we could do in the mean time is add support for
> auto-discovering all the plugins in a user's $HOME/.ginga/plugins folder
> and loading them. So that you don't need to use the --plugins parameter
> to load custom plugins if they are in that folder.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ejeschke/ginga/pull/146/files#r36943247>.
>